sklearn/datasets/_base.py

```diff
@@ -1454,28 +1464,160 @@ def _fetch_remote(remote, dirname=None, n_retries=3, delay=1):
 
     Returns
     -------
-    file_path: str
+    file_path: Path
         Full path of the created file.
     """
+    if dirname is None:
+        folder_path = Path(".")
+    else:
+        folder_path = Path(dirname)
+
+    file_path = folder_path / remote.filename
+
+    if file_path.exists():
+        if remote.checksum is None:
+            return file_path
+
+        checksum = _sha256(file_path)
+        if checksum == remote.checksum:
+            return file_path
+        else:
+            warnings.warn(
+                f"SHA256 checksum of existing local file {file_path.name} "
+                f"({checksum}) differs from expected ({remote.checksum}): "
+                f"re-downloading from {remote.url} ."
+            )
+
+    temp_file = NamedTemporaryFile(
```

Comment:
Just curious, here it seems like you are using `NamedTemporaryFile` only to get a filename, right?

Is it important to use `delete=False`, I guess maybe this is for Windows (glancing over the doc: https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile)?

Reply:
We need to close to be able to call `shutil.move`. I hadn't realized that `delete_on_close` existed in 3.12 but it's ignored when `delete=False`.

Not sure how to improve the comment. I think mentioning the 3.12 only `delete_on_close` option that we don't use would be more confusing than helpful.

Reply:
I decided to add a dedicated inline comment before the line where we close `temp_file` in anticipation: bed85156cdec5b257ed03771fce7344d6d194073.